### PR TITLE
Dont index groups (fixes distributed develop)

### LINKED ISSUE #2410: Several Opencast 10 Distributions Broken
Opencast fails with an OSGi wiring error when trying to start a worker distribution. My best guess is the restructured index rebuild service since that one is part of the resolution path that causes the error.

```
start-opencast: JAVA_HOME not set; results may vary
org.apache.felix.resolver.reason.ReasonException: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=opencast-worker; type=karaf.feature; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; filter:="(&(osgi.identity=opencast-worker)(type=karaf.feature)(version>=10.0.0.SNAPSHOT)(version<=10.0.0.SNAPSHOT))" [caused by: Unable to resolve opencast-worker/10.0.0.SNAPSHOT: missing requirement [opencast-worker/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-series-service-remote; type=osgi.bundle; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve opencast-series-service-remote/10.0.0.SNAPSHOT: missing requirement [opencast-series-service-remote/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.serviceregistry.api.ServiceRegistry)"; effective:=active [caused by: Unable to resolve opencast-serviceregistry/10.0.0.SNAPSHOT: missing requirement [opencast-serviceregistry/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.security.api.UserDirectoryService)"; effective:=active [caused by: Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement [opencast-userdirectory/10.0.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.opencastproject.index.rebuild)(version>=10.0.0)(!(version>=11.0.0)))"]]]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        at org.apache.felix.resolver.ResolverImpl.doResolve(ResolverImpl.java:392)
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:378)
        at org.apache.felix.resolver.ResolverImpl.resolve(ResolverImpl.java:332)
        at org.apache.karaf.features.internal.region.SubsystemResolver.resolve(SubsystemResolver.java:257)
        at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:393)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1062)
        at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:998)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-worker/10.0.0.SNAPSHOT: missing requirement [opencast-worker/10.0.0.SNAPSHOT] osgi.identity; osgi.identity=opencast-series-service-remote; type=osgi.bundle; version="[10.0.0.SNAPSHOT,10.0.0.SNAPSHOT]"; resolution:=mandatory [caused by: Unable to resolve opencast-series-service-remote/10.0.0.SNAPSHOT: missing requirement [opencast-series-service-remote/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.serviceregistry.api.ServiceRegistry)"; effective:=active [caused by: Unable to resolve opencast-serviceregistry/10.0.0.SNAPSHOT: missing requirement [opencast-serviceregistry/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.security.api.UserDirectoryService)"; effective:=active [caused by: Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement [opencast-userdirectory/10.0.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.opencastproject.index.rebuild)(version>=10.0.0)(!(version>=11.0.0)))"]]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 12 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-series-service-remote/10.0.0.SNAPSHOT: missing requirement [opencast-series-service-remote/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.serviceregistry.api.ServiceRegistry)"; effective:=active [caused by: Unable to resolve opencast-serviceregistry/10.0.0.SNAPSHOT: missing requirement [opencast-serviceregistry/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.security.api.UserDirectoryService)"; effective:=active [caused by: Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement [opencast-userdirectory/10.0.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.opencastproject.index.rebuild)(version>=10.0.0)(!(version>=11.0.0)))"]]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 13 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-serviceregistry/10.0.0.SNAPSHOT: missing requirement [opencast-serviceregistry/10.0.0.SNAPSHOT] osgi.service; filter:="(objectClass=org.opencastproject.security.api.UserDirectoryService)"; effective:=active [caused by: Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement [opencast-userdirectory/10.0.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.opencastproject.index.rebuild)(version>=10.0.0)(!(version>=11.0.0)))"]
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 14 more
Caused by: org.apache.felix.resolver.reason.ReasonException: Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement [opencast-userdirectory/10.0.0.SNAPSHOT] osgi.wiring.package; filter:="(&(osgi.wiring.package=org.opencastproject.index.rebuild)(version>=10.0.0)(!(version>=11.0.0)))"
        at org.apache.felix.resolver.Candidates$MissingRequirementError.toException(Candidates.java:1343)
        ... 15 more
```

Distributions that seems to be broken:

- Worker
- Presentation
- Ingest

Most likely cause:
> Unable to resolve opencast-userdirectory/10.0.0.SNAPSHOT: missing requirement […] org.o
pencastproject.index.rebuild […]

### PR DESCRIPTION
**Pull request by KatrinIhler**
I recently (?) broke distributed develop by making the userdirectory module depend directly on index modules that only exist on the admin node (while userdirectory is part of the opencast core).

Instead of ripping apart the userdirectory module to make this work, I decided that indexing groups doesn't make much sense anyway, since most institutions don't have that many groups, and we don't index the users right now, which there are probably more of.

So with this PR, we now get the groups directly from the database, and everything regarding group indexing in Elasticsearch is removed. Mostly everything works the same as before, with some minor exceptions:

- I deprecated sorting by roles and members of groups for the external API (the admin UI does not use this), since it doesn't make much sense and would have made the database query much more complicated. For versions < 1.6.0, the sorting by these fields happens manually afterwards. For this, I introduced a new API version 1.6.0, which might conflict with #2554. 
- I changed the behavior of the full text search (only available via admin UI) so that if you enter multiple words, _all_ of those words have to match a field in a group, instead of _any_ having to match. This is in violation of the other tables in the admin UI, but I think it makes much more sense and we should rather fix the others some time than re-implement their behavior.
- I no longer test user permissions, since that was broken before anyway - for non-admins, permissions were tested, but never added, so non-admins could not see groups (even ones they had created themselves) even if they have the correct GROUPS_VIEW role. Now this is only protected by the security config, which should be fine.

I also fixed the group creation in the frontend since it was just one line, now users get correctly added to new groups again.

This should fix #2410, but I didn't test this, maybe somebody else can?

### CODE REVIEW

**Review by lkiesow — APPROVED** [review]
I've verified that this fixes #2410 and correctly lists/filters the existing groups in the admin interface.

**Comment by gregorydlogan** [comment]
@KatrinIhler Does this require a reindex?  If so that should be added to the docs.  Also, something in the release notes.

**Review by gregorydlogan — APPROVED** [review]
I'm amused that you removed a ton of code, added very little, and managed to preserve the same basic behaviour.  Makes me wonder how useful indexing groups ever was... :)

I haven't tested this thoroughly, but it looks right and my basic tests pass.  Will try and get to this again next week, if someone else wants to merge this prior to that feel free.

**Comment by KatrinIhler** [comment]
> @KatrinIhler Does this require a reindex? If so that should be added to the docs. Also, something in the release notes.

I thought about this and I don't think it does? With this change, the groups will be directly picked from the database and new or updated groups won't be indexed, but it shouldn't matter whether you still have some old groups in the index, since they will never be queried, right? Of course a rebuild would be cleaner, but I don't think it's necessary, unless I missed something.

**Comment by KatrinIhler** [comment]
> I'm amused that you removed a ton of code, added very little, and managed to preserve the same basic behaviour. Makes me wonder how useful indexing groups ever was... :)

I think the answer you're looking for is: Not very. :D

**Comment by gregorydlogan** [comment]
> > @KatrinIhler Does this require a reindex? If so that should be added to the docs. Also, something in the release notes.
> 
> I thought about this and I don't think it does? With this change, the groups will be directly picked from the database and new or updated groups won't be indexed, but it shouldn't matter whether you still have some old groups in the index, since they will never be queried, right? Of course a rebuild would be cleaner, but I don't think it's necessary, unless I missed something.

I think you're probably right.  Ok, let's leave this then.

File: modules/admin-ui-index/src/main/resources/OSGI-INF/message-receiver-group.xml [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-               immediate="true" activate="activate" deactivate="deactivate">
-  <implementation
-    class="org.opencastproject.index.service.message.GroupMessageReceiverImpl"/>
-  <property name="service.description" value="Group Message Receiver"/>
-  <property name="destinationId" value="GROUP.Adminui"/>
-  <service>
-    <provide interface="org.opencastproject.index.service.message.GroupMessageReceiverImpl"/>
-  </service>
-
-  <reference name="message-broker-receiver"
```
Comment (lkiesow) [inline modules/admin-ui-index/src/main/resources/OSGI-INF/message-receiver-group.xml:13]: Is this the only module not on an admin node requiring the message broker modules? If so, we should remove them from the assemblies as well.

Comment (KatrinIhler) [reply modules/admin-ui-index/src/main/resources/OSGI-INF/message-receiver-group.xml:13]: Possible, but I don't have time to find that out right now. Can we leave this for later? :D

Comment (gregorydlogan) [reply modules/admin-ui-index/src/main/resources/OSGI-INF/message-receiver-group.xml:13]: The `admin-ui-index`, `scheduler-impl`, `external-api-index`, `conductor`, `live-schedule-impl`, `event-comment`, `authorization-manager`, and `themes` modules make OSGi references to `org.opencastproject.message.broker.api.MessageReceiver`.  A quick move of the message-broker modules into the admin* and allinone profiles yields booting builds.  I haven't had time to do anything other than make sure they start though...

Edit: filed at https://github.com/opencast/opencast/pull/2599

**Pull request merged**